staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_testing_utils_test.go

```diff
@@ -114,6 +113,11 @@ func compactStorage(c *Cacher) storagetesting.Compaction {
 		}
 		c.watchCache.listResourceVersion = rv
 
-		// TODO(wojtek-t): We should also compact the underlying etcd storage.
+		if _, err = client.KV.Put(ctx, "compact_rev_key", resourceVersion); err != nil {
```

Comment:
Don't like the fact that we are coding compaction here instead of just exposing Compact function and calling it.

Reply:
you mean using this function directly:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/etcd3/compact.go#L133
?

I would be fine with it if we can solve the circular dependency (although exposing Compact isn't perfect either...)

Reply:
I think this is good enough for now. This makes me even more eager to move compaction outside of storage, as it would clean this up.